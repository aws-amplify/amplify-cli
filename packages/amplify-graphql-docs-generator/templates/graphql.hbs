# this is an auto generated file. This will be overwritten
{{#each operations }}
  {{> renderOp }}
{{/each}}
{{#each fragments }}
  {{> renderExternalFragment }}
{{/each}}