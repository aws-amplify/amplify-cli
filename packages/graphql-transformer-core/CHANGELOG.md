# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.0.28"></a>
## [1.0.28](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.28-beta.0...graphql-transformer-core@1.0.28) (2018-10-18)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.28-beta.0"></a>
## [1.0.28-beta.0](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.12...graphql-transformer-core@1.0.28-beta.0) (2018-10-12)


### Bug Fixes

* **graphql-transformer-core:** Fix Readme.md docs for the `[@auth](https://github.com/auth)` directive ([db6ff7a](https://github.com/aws-amplify/amplify-cli/commit/db6ff7a))




<a name="1.0.12"></a>
## [1.0.12](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.11...graphql-transformer-core@1.0.12) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.11"></a>
## [1.0.11](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.10...graphql-transformer-core@1.0.11) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.10"></a>
## [1.0.10](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.9...graphql-transformer-core@1.0.10) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.9"></a>
## [1.0.9](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.8...graphql-transformer-core@1.0.9) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.8"></a>
## [1.0.8](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.7...graphql-transformer-core@1.0.8) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.7"></a>
## [1.0.7](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.6...graphql-transformer-core@1.0.7) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.6"></a>
## [1.0.6](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.5...graphql-transformer-core@1.0.6) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.5"></a>
## [1.0.5](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.4...graphql-transformer-core@1.0.5) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.4"></a>
## 1.0.4 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.3"></a>
## 1.0.3 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.2"></a>
## 1.0.2 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.1"></a>
## 1.0.1 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core
