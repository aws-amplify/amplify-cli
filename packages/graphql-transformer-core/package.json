{
  "name": "graphql-transformer-core",
  "version": "6.19.4",
  "description": "A framework to transform from GraphQL SDL to AWS CloudFormation.",
  "repository": {
    "type": "git",
    "url": "https://github.com/aws-amplify/amplify-cli.git",
    "directory": "packages/graphql-transformer-core"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "keywords": [
    "graphql",
    "cloudformation",
    "aws"
  ],
  "scripts": {
    "test": "jest",
    "test-ci": "jest --ci -i",
    "build": "tsc",
    "watch": "tsc -w",
    "clean": "rimraf ./lib"
  },
  "dependencies": {
    "cloudform-types": "^4.2.0",
    "deep-diff": "^1.0.2",
    "fs-extra": "^8.1.0",
    "glob": "^7.1.6",
    "graphql": "^14.5.8",
    "graphql-transformer-common": "4.17.4"
  },
  "devDependencies": {
    "@types/fs-extra": "^8.0.1",
    "@types/node": "^10.17.13"
  },
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "testURL": "http://localhost",
    "testRegex": "(src/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "collectCoverage": true
  }
}
