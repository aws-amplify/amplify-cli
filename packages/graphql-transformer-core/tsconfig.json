{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "strict": false, // TODO enable
    "rootDir": "src",
    "outDir": "lib"
  },
  "references": [
    {"path": "../graphql-transformer-common"}
  ]
}
