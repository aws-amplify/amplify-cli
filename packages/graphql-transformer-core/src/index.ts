import './polyfills/Object.assign';
import { print } from 'graphql';
import { TransformerContext } from './TransformerContext';
import { Transformer } from './Transformer';
import { ITransformer } from './ITransformer';
import { GraphQLTransform } from './GraphQLTransform';
import { collectDirectiveNames, collectDirectivesByTypeNames } from './collectDirectives';
import { stripDirectives } from './stripDirectives';
import { DeploymentResources } from './DeploymentResources';
import {
  buildProject as buildAPIProject,
  uploadDeployment as uploadAPIProject,
  migrateAPIProject,
  revertAPIMigration,
} from './util/amplifyUtils';
import {
  readSchema as readProjectSchema,
  loadProject as readProjectConfiguration,
  loadConfig as readTransformerConfiguration,
  writeConfig as writeTransformerConfiguration,
  TRANSFORM_CONFIG_FILE_NAME,
  TRANSFORM_BASE_VERSION,
  TRANSFORM_CURRENT_VERSION,
  TransformConfig,
  SyncConfig,
} from './util/transformConfig';
import { EXTRA_DIRECTIVES_DOCUMENT } from './validation';
export * from './errors';
export * from './util';

/**
 * Returns the extra set of directives that are supported by AppSync service
 */
export function getAppSyncServiceExtraDirectives(): string {
  return print(EXTRA_DIRECTIVES_DOCUMENT);
}

export {
  GraphQLTransform,
  TransformConfig,
  TransformerContext,
  Transformer,
  ITransformer,
  collectDirectiveNames,
  collectDirectivesByTypeNames,
  stripDirectives,
  buildAPIProject,
  migrateAPIProject,
  uploadAPIProject,
  readProjectSchema,
  readProjectConfiguration,
  readTransformerConfiguration,
  writeTransformerConfiguration,
  revertAPIMigration,
  TRANSFORM_CONFIG_FILE_NAME,
  TRANSFORM_BASE_VERSION,
  TRANSFORM_CURRENT_VERSION,
  SyncConfig,
  DeploymentResources,
};
