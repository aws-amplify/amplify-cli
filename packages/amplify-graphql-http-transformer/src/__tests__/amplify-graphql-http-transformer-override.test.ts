import { GraphQLTransform } from '@aws-amplify/graphql-transformer-core';
import { parse } from 'graphql';
import { HttpTransformer } from '..';
import path from 'path';
test('it generates the overrided resources', () => {
  const validSchema = /* GraphQL */ `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://www.api.com/ping")
      content2: String @http(method: PUT, url: "http://www.api.com/ping")
      more: String @http(url: "http://api.com/ping/me/2")
      evenMore: String @http(method: DELETE, url: "http://www.google.com/query/id")
      stillMore: String @http(method: PATCH, url: "https://www.api.com/ping/id")
    }
  `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
    overrideConfig: {
      overrideDir: path.join(__dirname, 'overrides'),
      overrideFlag: true,
      resourceName: 'myResource',
    },
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  expect(stack).toMatchSnapshot();
});
