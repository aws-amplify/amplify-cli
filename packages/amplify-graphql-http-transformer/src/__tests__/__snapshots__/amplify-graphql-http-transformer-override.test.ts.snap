// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it generates the overrided resources 1`] = `
Object {
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "An auto-generated nested stack for the @http directive.",
  "Parameters": Object {
    "referencetotransformerrootstackGraphQLAPI20497F53ApiId": Object {
      "Type": "String",
    },
    "referencetotransformerrootstackS3DeploymentBucket7592718ARef": Object {
      "Type": "String",
    },
    "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref": Object {
      "Type": "String",
    },
  },
  "Resources": Object {
    "Commentcontent2Resolver": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "FieldName": "content2",
        "Kind": "PIPELINE",
        "PipelineConfig": Object {
          "Functions": Array [
            Object {
              "Fn::GetAtt": Array [
                "httpwwwapicomDataSourceCommentcontent2FunctionhttpwwwapicomDataSourceCommentcontent2FunctionAppSyncFunctionA4F35A01",
                "FunctionId",
              ],
            },
          ],
        },
        "RequestMappingTemplate": "## [Start] Stash resolver specific context.. **
$util.qr($ctx.stash.put(\\"typeName\\", \\"Comment\\"))
$util.qr($ctx.stash.put(\\"fieldName\\", \\"content2\\"))
{}
## [End] Stash resolver specific context.. **",
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content2.res.vtl",
            ],
          ],
        },
        "TypeName": "Comment",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "CommentcontentResolver": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "FieldName": "content",
        "Kind": "PIPELINE",
        "PipelineConfig": Object {
          "Functions": Array [
            Object {
              "Fn::GetAtt": Array [
                "httpwwwapicomDataSourceCommentcontentFunctionhttpwwwapicomDataSourceCommentcontentFunctionAppSyncFunction685EFA2A",
                "FunctionId",
              ],
            },
          ],
        },
        "RequestMappingTemplate": "mockTemplate",
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content.res.vtl",
            ],
          ],
        },
        "TypeName": "Comment",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "CommentevenMoreResolver": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "FieldName": "evenMore",
        "Kind": "PIPELINE",
        "PipelineConfig": Object {
          "Functions": Array [
            Object {
              "Fn::GetAtt": Array [
                "httpwwwgooglecomDataSourceCommentevenMoreFunctionhttpwwwgooglecomDataSourceCommentevenMoreFunctionAppSyncFunction47773D49",
                "FunctionId",
              ],
            },
          ],
        },
        "RequestMappingTemplate": "## [Start] Stash resolver specific context.. **
$util.qr($ctx.stash.put(\\"typeName\\", \\"Comment\\"))
$util.qr($ctx.stash.put(\\"fieldName\\", \\"evenMore\\"))
{}
## [End] Stash resolver specific context.. **",
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.evenMore.res.vtl",
            ],
          ],
        },
        "TypeName": "Comment",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "CommentmoreResolver": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "FieldName": "more",
        "Kind": "PIPELINE",
        "PipelineConfig": Object {
          "Functions": Array [
            Object {
              "Fn::GetAtt": Array [
                "httpapicomDataSourceCommentmoreFunctionhttpapicomDataSourceCommentmoreFunctionAppSyncFunction5BEF6F35",
                "FunctionId",
              ],
            },
          ],
        },
        "RequestMappingTemplate": "## [Start] Stash resolver specific context.. **
$util.qr($ctx.stash.put(\\"typeName\\", \\"Comment\\"))
$util.qr($ctx.stash.put(\\"fieldName\\", \\"more\\"))
{}
## [End] Stash resolver specific context.. **",
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.more.res.vtl",
            ],
          ],
        },
        "TypeName": "Comment",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "CommentstillMoreResolver": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "FieldName": "stillMore",
        "Kind": "PIPELINE",
        "PipelineConfig": Object {
          "Functions": Array [
            Object {
              "Fn::GetAtt": Array [
                "httpswwwapicomDataSourceCommentstillMoreFunctionhttpswwwapicomDataSourceCommentstillMoreFunctionAppSyncFunction51FAE1A5",
                "FunctionId",
              ],
            },
          ],
        },
        "RequestMappingTemplate": "## [Start] Stash resolver specific context.. **
$util.qr($ctx.stash.put(\\"typeName\\", \\"Comment\\"))
$util.qr($ctx.stash.put(\\"fieldName\\", \\"stillMore\\"))
{}
## [End] Stash resolver specific context.. **",
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.stillMore.res.vtl",
            ],
          ],
        },
        "TypeName": "Comment",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "httpapicomDataSource": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "HttpConfig": Object {
          "Endpoint": "mockEndpoint",
        },
        "Name": "httpapicomDataSource",
        "ServiceRoleArn": "mockArn",
        "Type": "HTTP",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "httpapicomDataSourceCommentmoreFunctionhttpapicomDataSourceCommentmoreFunctionAppSyncFunction5BEF6F35": Object {
      "DependsOn": Array [
        "httpapicomDataSource",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "DataSourceName": Object {
          "Fn::GetAtt": Array [
            "httpapicomDataSource",
            "Name",
          ],
        },
        "FunctionVersion": "2018-05-29",
        "Name": "httpapicomDataSourceCommentmoreFunction",
        "RequestMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.more.DataResolver.req.vtl",
            ],
          ],
        },
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.more.DataResolver.res.vtl",
            ],
          ],
        },
      },
      "Type": "AWS::AppSync::FunctionConfiguration",
    },
    "httpapicomDataSourceServiceRoleC12170E3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "appsync.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "httpswwwapicomDataSource": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "HttpConfig": Object {
          "Endpoint": "mockEndpoint",
        },
        "Name": "httpswwwapicomDataSource",
        "ServiceRoleArn": "mockArn",
        "Type": "HTTP",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "httpswwwapicomDataSourceCommentstillMoreFunctionhttpswwwapicomDataSourceCommentstillMoreFunctionAppSyncFunction51FAE1A5": Object {
      "DependsOn": Array [
        "httpswwwapicomDataSource",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "DataSourceName": Object {
          "Fn::GetAtt": Array [
            "httpswwwapicomDataSource",
            "Name",
          ],
        },
        "FunctionVersion": "2018-05-29",
        "Name": "httpswwwapicomDataSourceCommentstillMoreFunction",
        "RequestMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.stillMore.DataResolver.req.vtl",
            ],
          ],
        },
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.stillMore.DataResolver.res.vtl",
            ],
          ],
        },
      },
      "Type": "AWS::AppSync::FunctionConfiguration",
    },
    "httpswwwapicomDataSourceServiceRoleDE26F49B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "appsync.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "httpwwwapicomDataSource": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "HttpConfig": Object {
          "Endpoint": "mockEndpoint",
        },
        "Name": "httpwwwapicomDataSource",
        "ServiceRoleArn": "mockArn",
        "Type": "HTTP",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "httpwwwapicomDataSourceCommentcontent2FunctionhttpwwwapicomDataSourceCommentcontent2FunctionAppSyncFunctionA4F35A01": Object {
      "DependsOn": Array [
        "httpwwwapicomDataSource",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "DataSourceName": Object {
          "Fn::GetAtt": Array [
            "httpwwwapicomDataSource",
            "Name",
          ],
        },
        "FunctionVersion": "2018-05-29",
        "Name": "httpwwwapicomDataSourceCommentcontent2Function",
        "RequestMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content2.DataResolver.req.vtl",
            ],
          ],
        },
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content2.DataResolver.res.vtl",
            ],
          ],
        },
      },
      "Type": "AWS::AppSync::FunctionConfiguration",
    },
    "httpwwwapicomDataSourceCommentcontentFunctionhttpwwwapicomDataSourceCommentcontentFunctionAppSyncFunction685EFA2A": Object {
      "DependsOn": Array [
        "httpwwwapicomDataSource",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "DataSourceName": Object {
          "Fn::GetAtt": Array [
            "httpwwwapicomDataSource",
            "Name",
          ],
        },
        "FunctionVersion": "2018-05-29",
        "Name": "httpwwwapicomDataSourceCommentcontentFunction",
        "RequestMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content.DataResolver.req.vtl",
            ],
          ],
        },
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.content.DataResolver.res.vtl",
            ],
          ],
        },
      },
      "Type": "AWS::AppSync::FunctionConfiguration",
    },
    "httpwwwapicomDataSourceServiceRole20167D8C": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "appsync.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "httpwwwgooglecomDataSource": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "HttpConfig": Object {
          "Endpoint": "mockEndpoint",
        },
        "Name": "httpwwwgooglecomDataSource",
        "ServiceRoleArn": "mockArn",
        "Type": "HTTP",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "httpwwwgooglecomDataSourceCommentevenMoreFunctionhttpwwwgooglecomDataSourceCommentevenMoreFunctionAppSyncFunction47773D49": Object {
      "DependsOn": Array [
        "httpwwwgooglecomDataSource",
      ],
      "Properties": Object {
        "ApiId": Object {
          "Ref": "referencetotransformerrootstackGraphQLAPI20497F53ApiId",
        },
        "DataSourceName": Object {
          "Fn::GetAtt": Array [
            "httpwwwgooglecomDataSource",
            "Name",
          ],
        },
        "FunctionVersion": "2018-05-29",
        "Name": "httpwwwgooglecomDataSourceCommentevenMoreFunction",
        "RequestMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.evenMore.DataResolver.req.vtl",
            ],
          ],
        },
        "ResponseMappingTemplateS3Location": Object {
          "Fn::Join": Array [
            "",
            Array [
              "s3://",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentBucket7592718ARef",
              },
              "/",
              Object {
                "Ref": "referencetotransformerrootstackS3DeploymentRootKeyA71EA735Ref",
              },
              "/resolvers/Comment.evenMore.DataResolver.res.vtl",
            ],
          ],
        },
      },
      "Type": "AWS::AppSync::FunctionConfiguration",
    },
    "httpwwwgooglecomDataSourceServiceRole08261832": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "appsync.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
